src/util/system.cpp

```diff
@@ -940,6 +940,12 @@ bool IsConfSupported(KeyInfo key, std::string& error) {
         error = "conf cannot be set in the configuration file; use includeconf= if you want to include additional config files";
         return false;
     }
+    if (key.name == "reindex") {
+        // reindex can be set in a config file but it is strongly discouraged as this will cause the node to reindex on
+        // every restart. Allow the config but throw a warning
+        LogPrintf("Warning: reindex is set which will cause this node to reindex on every restart. Consider using -reindex=1 from the command-line\n");
```

Comment:
In commit "util: warn if reindex is used in conf" (5cf1b465f4f9352415604c3eb31fc8cd01e4bd46)

This wording is not good because it doesn't clearly state the cause of the problem. It makes it sound like just setting reindex in general causes reindexing every restart. Also it makes it sound like adding -reindex=1 to the command line would fix the problem, when it wouldn't and the actual fix is to *remove* reindex=1 from the config file.

Would suggest: "Warning: reindex=1 is set in the configuration file, which will significantly slow down startup. Consider removing or commenting out this option for better performance, unless there is currently a condition which makes rebuilding the indexes necessary."

Reply:
nice, i like the suggested wording much better.